### PR TITLE
filter nil elements from projetile-paths-to-ignore

### PR DESCRIPTION
**Pull request by cowboyd**
This applies the fix suggested by @bogus34 in #633

### CODE REVIEW

File: projectile.el [hunk 1]
```diff
@@ -1026,9 +1026,9 @@ Only buffers not visible in windows are returned."
 
 (defun projectile-paths-to-ignore ()
   "Return a list of ignored project paths."
-  (--mapcat (and (string-prefix-p "/" it)
-                 (directory-file-name it))
-            (cdr (projectile-parse-dirconfig-file))))
+  (-non-nil (--map (and (string-prefix-p "/" it)
+                     (directory-file-name it))
```
Comment (bbatsov) [inline projectile.el]: This is not indented properly.

**Comment by bbatsov** [comment]
I've tried this locally and ignoring still doesn't work. The error is gone, but nothing is actually ignored...

File: projectile.el (hunk 1 above)
Comment (cowboyd) [inline projectile.el]: hmm.. That's weird. I have no customizations on elisp or lisp indentation, although I did recently update prelude. 

Do you know offhand where these settings are? I can't seem to find any customizations for elisp indentation.

**Comment by cowboyd** [comment]
I also noticed this behavior last night. Apologies for not verifying before submitting the request.

File: projectile.el (hunk 1 above)
Comment (cowboyd) [inline projectile.el]: nvm. when I never re-indented after copying the patch from @bogus34

**Comment by cowboyd** [comment]
```
-mapcat is a compiled Lisp function in `dash.el'.

(-mapcat FN LIST)

Return the concatenation of the result of mapping FN over LIST.
Thus function FN should return a list.
```

Just wrapping the directory name in the mapping function in a list does appear to work. If this seems like a reasonable approach, I can wrap this with a test to ensure the correct behavior.

**Comment by cowboyd** [comment]
I believe this is closed by  41424b9fe2f1e12c3f2abbbad3d866ec23bd995d

**Pull request closed without merging**